diff --git a/home/mrdrivingduck/Desktop/d4jDataBackup/Math_98/src/java/org/apache/commons/math/linear/BigMatrixImpl.java b/1_BigMatrixImpl.java
index 7cd61cb..81214ae 100644
--- a/home/mrdrivingduck/Desktop/d4jDataBackup/Math_98/src/java/org/apache/commons/math/linear/BigMatrixImpl.java
+++ b/1_BigMatrixImpl.java
@@ -986,9 +986,16 @@ public class BigMatrixImpl implements BigMatrix, Serializable {
         if (v.length != this.getColumnDimension()) {
             throw new IllegalArgumentException("vector has wrong length");
         }
+// start of generated patch
+ final int nRows=this.getRowDimension();
+ final int nCols=this.getColumnDimension();
+ final BigDecimal[] out=new BigDecimal[nRows];
+// end of generated patch
+/* start of original code
         final int nRows = this.getRowDimension();
         final int nCols = this.getColumnDimension();
         final BigDecimal[] out = new BigDecimal[v.length];
+ end of original code*/
         for (int row = 0; row < nRows; row++) {
             BigDecimal sum = ZERO;
             for (int i = 0; i < nCols; i++) {
