diff --git a/home/mrdrivingduck/Desktop/d4jDataBackup/Math_85/src/java/org/apache/commons/math/analysis/solvers/UnivariateRealSolverUtils.java b/1_UnivariateRealSolverUtils.java
index bf3e4bf..4173554 100644
--- a/home/mrdrivingduck/Desktop/d4jDataBackup/Math_85/src/java/org/apache/commons/math/analysis/solvers/UnivariateRealSolverUtils.java
+++ b/1_UnivariateRealSolverUtils.java
@@ -195,6 +195,12 @@ public class UnivariateRealSolverUtils {
         } while ((fa * fb > 0.0) && (numIterations < maximumIterations) && 
                 ((a > lowerBound) || (b < upperBound)));
    
+// start of generated patch
+if(fa*upperBound>=0.0){
+throw new ConvergenceException("number of iterations={0}, maximum iterations={1}, "+"initial={2}, lower bound={3}, upper bound={4}, final a value={5}, ",numIterations,maximumIterations,initial,lowerBound,upperBound,a,b,fa,fb);
+}
+// end of generated patch
+/* start of original code
         if (fa * fb >= 0.0 ) {
             throw new ConvergenceException(
                       "number of iterations={0}, maximum iterations={1}, " +
@@ -203,6 +209,7 @@ public class UnivariateRealSolverUtils {
                       numIterations, maximumIterations, initial,
                       lowerBound, upperBound, a, b, fa, fb);
         }
+ end of original code*/
         
         return new double[]{a, b};
     }
