diff --git a/home/mrdrivingduck/Desktop/d4jDataBackup/Math_59/src/main/java/org/apache/commons/math/util/FastMath.java b/1_FastMath.java
index 8cba4d4..479ad24 100644
--- a/home/mrdrivingduck/Desktop/d4jDataBackup/Math_59/src/main/java/org/apache/commons/math/util/FastMath.java
+++ b/1_FastMath.java
@@ -3479,7 +3479,12 @@ public class FastMath {
      * @return b if a is lesser or equal to b, a otherwise
      */
     public static float max(final float a, final float b) {
+// start of generated patch
+return (a<=b)?b:(Float.isNaN(a+b)?Float.NaN:a);
+// end of generated patch
+/* start of original code
         return (a <= b) ? b : (Float.isNaN(a + b) ? Float.NaN : b);
+ end of original code*/
     }
 
     /** Compute the maximum of two values
