diff --git a/home/mrdrivingduck/Desktop/d4jDataBackup/Math_35/src/main/java/org/apache/commons/math3/genetics/ElitisticListPopulation.java b/1_ElitisticListPopulation.java
index 67b368d..555ecc5 100644
--- a/home/mrdrivingduck/Desktop/d4jDataBackup/Math_35/src/main/java/org/apache/commons/math3/genetics/ElitisticListPopulation.java
+++ b/1_ElitisticListPopulation.java
@@ -61,8 +61,17 @@ public class ElitisticListPopulation extends ListPopulation {
      * @throws OutOfRangeException if the elitism rate is outside the [0, 1] range
      */
     public ElitisticListPopulation(final int populationLimit, final double elitismRate) {
+// start of generated patch
+super(populationLimit);
+if(elitismRate<0||elitismRate>1){
+throw new OutOfRangeException(LocalizedFormats.ELITISM_RATE,elitismRate,0,1);
+}
+this.elitismRate=elitismRate;
+// end of generated patch
+/* start of original code
         super(populationLimit);
         this.elitismRate = elitismRate;
+ end of original code*/
     }
 
     /**
